server/server.go

```diff
@@ -157,6 +157,18 @@ func (s *Server) StartEtcd(apiHandler http.Handler) error {
 	s.client = client
 	s.id = uint64(etcd.Server.ID())
 
+	// update advertise peer urls.
+	etcdMembers, err := listEtcdMembers(client)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, m := range etcdMembers.Members {
+		if s.ID() == m.ID {
+			s.cfg.AdvertisePeerUrls = strings.Join(m.PeerURLs, ",")
```

Comment:
Etcd's name and client-urls is passed from arguments, so there is no need to update those.